Calendar/CalLogHelpers/CalLogExportFunctions.ps1

```diff
@@ -0,0 +1,302 @@
+    $script:GCDO | Export-Csv -Path $FilenameRaw -NoTypeInformation -Encoding UTF8
+}
+
+# Export to Excel
+function Export-CalLogExcel {
+    Write-Host -ForegroundColor Cyan "Exporting Enhanced CalLogs to Excel Tab [$ShortId]..."
+    $ExcelParamsArray = GetExcelParamsArray -path $FileName -tabName $ShortId
+
+    $excel = $GCDOResults | Export-Excel @ExcelParamsArray -PassThru
+
+    FormatHeader ($excel)
+
+    Export-Excel -ExcelPackage $excel -WorksheetName $ShortId -MoveToStart
+
+    # Export Raw Logs for Developer Analysis
+    Write-Host -ForegroundColor Cyan "Exporting Raw CalLogs to Excel Tab [$($ShortId + "_Raw")]..."
+    $script:GCDO | Export-Excel -Path  $FileName -WorksheetName $($ShortId + "_Raw") -AutoFilter -FreezeTopRow -BoldTopRow -MoveToEnd
+}
+
+function Export-Timeline {
+    Write-Host -ForegroundColor Cyan "Exporting Timeline to file..."
+    Write-Host -ForegroundColor Green "Export to Excel is : $ExportToExcel"
+
+    # Display Timeline to screen:
+    Write-Host -ForegroundColor Cyan "Timeline for [$Identity]..."
+    $script:TimeLineOutput
+
+    if ($ExportToExcel.IsPresent) {
+        Export-TimelineExcel
+    } else {
+        $script:TimeLineOutput | Export-Csv -Path $script:TimeLineFilename -NoTypeInformation -Encoding UTF8 -Append
+    }
+}
+
+function Export-TimelineExcel {
+    Write-Host -ForegroundColor Cyan "Exporting Timeline to Excel..."
+    $script:TimeLineOutput | Export-Excel -Path $FileName -WorksheetName $($ShortId + "_TimeLine") -Title "Timeline for $Identity" -AutoSize -FreezeTopRow -BoldTopRow
+}
+
+function GetExcelParamsArray($path, $tabName) {
+    if ($script:IsOrganizer) {
+        $TableStyle = "Light10" # Orange for Organizer
+        $TitleExtra = ", Organizer"
+    } elseif ($script:IsRoomMB) {
+        Write-Host -ForegroundColor green "Room Mailbox Detected"
+        $TableStyle = "Light11" # Green for Room Mailbox
+        $TitleExtra = ", Resource"
+    } else {
+        $TableStyle = "Light12" # Light Blue for normal
+        # Dark Blue for Delegates (once we can determine this)
+    }
+
+    return @{
+        Path                    = $path
+        FreezeTopRow            = $true
+        #  BoldTopRow              = $true
+        Verbose                 = $false
+        TableStyle              = $TableStyle
+        WorksheetName           = $tabName
+        TableName               = $tabName
+        FreezeTopRowFirstColumn = $true
+        AutoFilter              = $true
+        AutoNameRange           = $true
+        Append                  = $true
+        Title                   = "Enhanced Calendar Logs for $Identity" + $TitleExtra + " for MeetingID [$($script:GCDO[0].CleanGlobalObjectId)]."
+        ConditionalText         = $ConditionalFormatting
+    }
+}
+
+#Sample
+#ConditionalText : @{Text=Outlook; ConditionalTextColor=Color [DarkRed]; ConditionalType=ContainsText; PatternType=Solid; Range=; BackgroundColor=Color [LightPink]}
+
+$ConditionalFormatting = $(
```

Comment:
For code readability, I like having all the Condition Text in its own section. I am guessing that it will grow. 
I could make this a constant or something, if you think would help.

Reply:
Not sure if this reads better, because it would virtually be the same it would just be created directly within the object that we want those values for.

Reply:
Unless you push back, I am going to leave it as is for now. 
Rui is looking at storing the formatting values another way so that they can be imported / exported, etc. so we will revisit this then.

Reply:
Sounds good